### PR TITLE
Added typings for the gaugejs package.

### PR DESCRIPTION
**Pull request by BTOdell**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by BTOdell** [comment]
There really should be some documentation in the README that specifies the new directory name must be lowercase even if the package on NPM has uppercase characters, and how to work around the npm-name check performed by the linter.

I got screwed by the Windows file system because it's case-insensitive and git couldn't detect that I had renamed the directory, so it left the old "gaugeJS" directory in the index and the build kept failing. 😰

**Comment by BTOdell** [comment]
For some reason the Travis integration report keeps getting stuck even though it shows the build was successful on the travis-ci.org website.
Can someone look into this? Or just accept this pull request because everything is good to go.

File: types/gaugejs/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/gaugejs/tslint.json:3]: @sandersn @orta do you think its ok to disable this rule or do we need something special here
The npm package name is not `gaugejs` https://www.npmjs.com/package/gaugeJS

Comment (orta) [reply types/gaugejs/tslint.json:3]: I don't know enough, will defer to Nathan 👍

Comment (BTOdell) [reply types/gaugejs/tslint.json:3]: I started by naming the types directory using the exact name "gaugeJS" but then the DefinitelyTyped CI build errored out saying that the package name must be all lowercase.
There was no documentation on how to properly handle this case so I just tried different workarounds until I got the CI build passing.

**Comment by sandersn** [comment]
Unfortunately, since late 2017, [npm doesn't allow new packages to use uppercase letters](https://blog.npmjs.org/post/168978377570/new-package-moniker-rules), so the @types package will have to mismatch the actual package name. Disabling the npm-naming rule is the right thing in this case. You could also add "non-npm package" to the first line of the header:

```ts
// Type definitions for non-npm package gaugeJS 1.3
```

But this is not really true.

**Comment by BTOdell** [comment]
@sandersn I just started using these type definitions (for gaugeJS) in my own project and the TypeScript compiler can't find the typings when compiling on Linux but it compiles fine on Windows. I suspect this is because Windows has a case-insensitive file system.

Is there a way to update the typings in the DefinitelyTyped repository so they will be used when importing "gaugeJS" in my own project? Currently, if I import "gaugejs" it finds the typings but it fails at runtime, and if I import "gaugeJS" it produces a compile error but still emits and runs without error.

**Comment by sandersn** [comment]
@BTODell can you try wrapping the entire .d.ts file with `declare module "gaugeJS" { ... }` ? That will require some finagling to get past CI here, but you might be able to test whether it works locally first.

**Comment by sandersn** [comment]
Just tested it locally, and it works for me. I just had to delete some `declare` keywords because our rules about when `declare` is allowed are stupid.

**Comment by sandersn** [comment]
(Feel free to disable as many lint rules as you need to get this fix merged =)

**Comment by BTOdell** [comment]
@sandersn Yes, that is what I have been doing as a workaround until I can figure out to get the types working properly using the DefinitelyTyped repo. I just have a local copy of the .d.ts file that I've wrapped with `declare module "gaugeJS" { ... }`

If I apply the same "workaround" directly in the DefinitelyTyped repo, will TypeScript be able to find the declared module definition if it's still deployed under the "@types/gaugejs" package? If I'm importing the library as "gaugeJS", I would assume the TypeScript compiler wouldn't know to even look in the "@types/gaugejs" package for other module definitions.

**Pull request merged**